server/sonar-server/src/main/java/org/sonar/server/db/DatabaseChecker.java

```diff
@@ -58,16 +59,26 @@ public void stop() {
     // nothing to do
   }
 
-  private void checkOracleDriverVersion() throws SQLException {
+  private void checkOracleVersion() throws SQLException {
     Connection connection = db.getDataSource().getConnection();
     try {
+      // check version of db
+      // See http://jira.codehaus.org/browse/SONAR-6434
+      int majorVersion = connection.getMetaData().getDatabaseMajorVersion();
+      if (majorVersion < 11) {
```

Comment:
I would extract the 11 in a constant (no magic numbers) to ease next compatibility break.